src/App.js

```diff
@@ -1,23 +1,33 @@
-import React from 'react';
+import React from "react";
 
-import { getAll, getById } from './api/phone'
-import Basket from './Basket'
-import Filter from './Filter'
-import Catalog from './Catalog'
-
-import './App.css';
+import { getAll, getById } from "./api/phone";
+import Basket from "./Basket";
+import Filter from "./Filter";
+import Catalog from "./Catalog";
 
+import "./App.css";
+// import { isTerminatorless } from "@babel/types";
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
+  state = {
+    phones: getAll(),
+    selectedPhone: null,
+    basketItems: []
+  };
 
-    this.state = {
-      phones: getAll(),
-      selectedPhone: null,
-      basketItems: [],
-    };
-  }
+  addToCart = (name, id) => {
+    this.setState(prevState => ({
+      basketItems: [...prevState.basketItems, { name, id }]
+    }));
+  };
+
+  removeFromCart = name => {
+    this.setState({
+      basketItems: this.state.basketItems.filter(item =>
```

Comment:
`Array.prototype.filter()` takes function which returns `true` or `false`. You don't need to return `item` because it's casted as boolean => `true`
```suggestion
      basketItems: this.state.basketItems.filter(item => item.name !== name)
```

Reply:
Now if I add 5 items and click `remove` on some of them all 5 items are being removed. It should decrease count of item and remove item only if count = 1

Reply:
> Now if I add 5 items and click remove on some of them all 5 items are being removed. It should decrease count of item and remove item only if count = 1

This was not fixed